OptimoveSDK/Sources/Classes/PreferenceCenter/OptimovePreferenceCenter.swift

```diff
@@ -0,0 +1,191 @@
+import Foundation
+import OptimoveCore
+
+@available(iOS 13.0, *)
+public class OptimovePreferenceCenter {
+    enum Error: LocalizedError {
+        case alreadyInitialized
+        case configurationIsMissing
+
+        var errorDescription: String? {
+            switch self {
+            case .alreadyInitialized:
+                return "The PreferenceCenterSDK has already been initialized."
+            case .configurationIsMissing:
+                return "Preference center config is missing."
+            }
+        }
+    }
+
+    private static var instance: OptimovePreferenceCenter?
+    private var networkClient: NetworkClient?
+    private var storage: OptimoveStorage?
+
+    static var isSdkRunning: Bool {
+        return Optimove.getConfig()?.getPreferenceCenterConfig() != nil
+    }
+
+    public enum ResultType {
+        case success
+        case errorUserNotSet
+        case error
+    }
+
+    public typealias PreferencesGetHandler = (_ result: ResultType, _ preferences: Preferences?) -> Void
+    public typealias PreferencesSetHandler = (_ result: ResultType) -> Void
+
+    public static func getInstance() throws -> OptimovePreferenceCenter {
+        guard let instance = instance else {
+            throw Error.configurationIsMissing
+        }
+        return instance
+    }
+
+    static func initialize(storage: OptimoveStorage, networkClient: NetworkClient) throws {
+        guard instance == nil else {
```

Comment:
1. we do delayed init, we call Optimove.initialize, which calls OptimovePreferenceCenter.initialize
2. then we do setCredentials, which calls initialize(with: config), which calls OptimovePreferenceCenter.initialize
3. it throws an error here, which we log
----> every delayed init of PC logs an error

moral
1. this happens because android and ios work slightly differently. On android with delayed init we dont call Optimove.initialize again, on ios we do. With such little differences it becomes harder to verify 2 logics -- to me having them as close as possible is ideal, even when it sometimes deviates from language best practices (but ofc still within language acceptable practices :) ) 
2. we should not only make working code, but also care about best possible API / error logging -- customer experience -- which is a direct source of inquiries. This involves deciding what we should log and how, actually checking that correct errors logged, having sensible error messages etc. Just feels logging is not getting enough attention over and over again, here and in android sdk and in general

Reply:
Agree with this, but not sure what to do to resolve this.

Are you suggesting removing the `Logger.error("Could not find preference center credentials")` below and just returning `nil`? That way we just won't proceed with pref center initialisation and won't notify anyone? Or another solution? @vvoicehovics

Reply:
can check what other features do, e.g. optimobile

```
    static func initialize(config optimoveConfig: OptimoveConfig, initialVisitorId: String, initialUserId: String?) throws {
        if instance !== nil, optimoveConfig.features.contains(.delayedConfiguration) {
            try completeDelayedConfiguration(config: optimoveConfig.optimobileConfig!)
            return
        }
```

so
a) if it's already initialized and it is a delayed init -- skip
b) even simpler, if it's already initialized -- just dont do anything. credentials dont affect initialisation